langchain/document_loaders/pdf.py

```diff
@@ -381,3 +381,91 @@ def load(self) -> List[Document]:
             contents = self.clean_pdf(contents)
         metadata = {"source": self.source, "file_path": self.source}
         return [Document(page_content=contents, metadata=metadata)]
+
+
+class PDFPlumberLoader(BasePDFLoader):
+    """Loader that uses PDFPlumber to load PDF files."""
+
+    def __init__(self, file_path: str):
+        """Initialize with file path."""
+        try:
+            import pdfplumber  # noqa:F401
+        except ImportError:
+            raise ValueError(
+                "PDFPlumber package not found, please install it with "
+                "`pip install pdfplumber`"
+            )
+
+        super().__init__(file_path)
+
+    def load(self, **kwargs: Optional[Any]) -> List[Document]:
```

Comment:
The parent class defines a `load` method which does not allow args or kwargs.


```suggestion
    def load(self) -> List[Document]:
```

kwargs can be provided from the __ init __ instead.

The reason to keep the load method consistent is to follow the following rule of thumb for sub-classes: https://en.wikipedia.org/wiki/Liskov_substitution_principle

This loader looks a lot like the pymypdf loader. Is pdfplumber using pymupdf under the hood?

Reply:
pdfplumber looks to be build on [pdfminer.six](https://github.com/pdfminer/pdfminer.six), and the documentation provides a overview of [comparisons](https://github.com/jsvine/pdfplumber#comparison-to-other-libraries) with other PDF parsing libraries.

For passing the kwargs, I thought we can update class dictionary with `self.__dict__.update(kwargs)` after `__init__` with **kwargs, but then the different methods of pdfplumber would not be able to access without explicit `get` on the class dictionary. And the kwargs can get extensive...
```
.extract_words(x_tolerance=3, y_tolerance=3, keep_blank_chars=False, use_text_flow=False, horizontal_ltr=True, vertical_ttb=True, extra_attrs=[], split_at_punctuation=False, expand_ligatures=True)
```
Again, maybe there's a better way.